tests/test_daemon.py

```diff
@@ -0,0 +1,258 @@
+        (
+            "# Some comment\n"
+            "Date,Time,Port,DB,InputType,OutputDir,"
+            "PID,IsDaemon\n"
+            "2023-07-25,10:00:00,6379,redis,pcap,"
+            "/tmp/output,12345,True\n",
+            ("6379", "/tmp/output", "12345"),
+        ),
+        # Test case 2: Multiple entries, last one is daemon
+        (
+            "2023-07-25,09:00:00,6380,redis,pcap,"
+            "/tmp/output1,12344,False\n"
+            "2023-07-25,10:00:00,6379,redis,pcap,"
+            "/tmp/output2,12345,True\n",
+            ("6379", "/tmp/output2", "12345"),
+        ),
+        # Test case 3: Empty file
+        ("", None),
+    ],
+)
+def test_get_last_opened_daemon_info(file_content, expected_result):
+    daemon = ModuleFactory().create_daemon_object()
+    daemon.slips.redis_man.running_logfile = "mock_logfile.txt"
+
+    with patch("builtins.open", mock_open(read_data=file_content)):
+        result = daemon.get_last_opened_daemon_info()
+
+    assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    "pidfile_exists, expected_output, " "expected_remove_calls",
+    [
+        # Test case 1: pidfile exists and is deleted
+        (True, ["pidfile deleted."], [call("/tmp/slips_daemon.lock")]),
+        # Test case 2: pidfile doesn't exist
+        (
+            False,
+            [
+                "Can't delete pidfile, /tmp/slips_daemon.lock doesn't exist.",
+                "Either Daemon stopped normally or an error occurred.",
+            ],
+            [],
+        ),
+    ],
+)
+def test_delete_pidfile(
+    pidfile_exists, expected_output, expected_remove_calls
+):
+    daemon = ModuleFactory().create_daemon_object()
+    with patch("os.path.exists", return_value=pidfile_exists), patch(
+        "os.remove"
+    ) as mock_remove, patch.object(daemon, "print") as mock_print:
+        daemon.delete_pidfile()
+
+        mock_remove.assert_has_calls(expected_remove_calls)
+        mock_print.assert_has_calls([call(line) for line in expected_output])
+
+
+@pytest.mark.parametrize(
+    "pid, os_kill_side_effect",
+    [
+        # Test case 1: Successfully kill the daemon
+        (12345, None),
+        # Test case 2: Daemon already killed
+        (12345, ProcessLookupError),
+    ],
+)
+def test_killdaemon(pid, os_kill_side_effect):
+    daemon = ModuleFactory().create_daemon_object()
+    daemon.pid = str(pid)
+
+    with patch("os.kill", side_effect=os_kill_side_effect) as mock_kill:
+        daemon.killdaemon()
+
+    mock_kill.assert_called_once_with(pid, signal.SIGTERM)
+
+
+@pytest.mark.parametrize(
+    "pid, lock_side_effect," " expected_result",
+    [
+        # Testcase1:pid_exists_lock_acquired
+        (12345, None, True),
+        # Testcase2:no_pid_lock_acquired
+        (None, None, False),
+        # Testcase3:pid_exists_lock_not_acquired
+        (12345, CannotAcquireLock(), True),
+        # Testcase4:no_pid_lock_not_acquired
+        (None, CannotAcquireLock(), False),
+    ],
+)
```

Comment:
comments in the parametrize() are written in a weird way, it's better to make them space separated